lib/srv/db/server.go

```diff
@@ -358,6 +435,13 @@ func (s *Server) startDatabase(ctx context.Context, database types.Database) err
 		// Log, but do not fail. We will fetch the version later.
 		s.log.Warnf("Failed to fetch the MySQL version for %s: %v", database.GetName(), err)
 	}
+
+	if database.GetProtocol() == defaults.ProtocolSQLServer {
+		if err := s.startSQLServerLDAP(ctx, database); err != nil {
```

Comment:
@alistanis 
The startSQLServerLDAP functions calls
`startSQLServerLDAP-> InitializeLDAP-> tlsConfigForLDAP-> GenerateCredentials` 

where [GenerateCredentials](https://github.com/gravitational/teleport/blob/6d74ef49fdf079c33c7cd1e0f9a3eb5c354f9009/lib/auth/windows/windows.go#L115)
does the `authClient.GenerateWindowsDesktopCert` call 

How does the DB service `startSQLServerLDAP` call works if `GenerateWindowsDesktopCert` endpoint allows to be invoked only by `RoleWindowsDesktop` role [link](https://github.com/gravitational/teleport/blob/dd67a49ab5035e51847a3fc8934f53b5abe25a6a/lib/auth/auth_with_roles.go#L5060) ?